tools/helpers.ts

```
@@ -0,0 +1,239 @@
+  for (const tok in dictionary) {
+    text = text.replace(new RegExp(escapeRegExp(tok), 'g'), dictionary[tok]);
+  }
+  return text;
+}
+
+/**
+ * Reads a template file, inflates it with the provided dictionary, and writes the prettified
+ * result to destination file
+ *
+ * @param templatePath path to template file
+ * @param dest destination file
+ * @param dict dictionary to inflate the template
+ */
+export async function inflateTemplateToFile(
+  templatePath: string,
+  dest: string,
+  dict: { [x: string]: string },
+) {
+  const template = await readFile(templatePath, 'utf8');
+  await writePrettyFile(dest, inflateTemplate(template, dict));
+}
+
+/**
+ * From the RSPEC json file, creates a generated-meta.ts file with ESLint formatted metadata
+ *
+ * @param sonarKey rule ID for which we need to create the generated-meta.ts file
+ */
+export async function generateMetaForRule(sonarKey: string) {
+  const rspecFile = join(METADATA_FOLDER, `${sonarKey}.json`);
+  const rspecFileExists = await exists(rspecFile);
+  const ruleRspecMeta: rspecMeta = rspecFileExists
+    ? JSON.parse(await readFile(rspecFile, 'utf-8'))
+    : {
+        // Dummy data to create compilable new rule metadata
+        title: 'Description',
+        tags: [],
+        type: 'BUG',
+        status: 'ready',
+        quickfix: 'covered',
+      };
+
+  if (!rspecFileExists) {
+    console.log(
+      `RSPEC metadata not found for rule ${sonarKey}. Creating dummy "generated-meta.ts"`,
+    );
+  }
+
+  if (!typeMatrix[ruleRspecMeta.type]) {
+    console.log(`Type not found for rule ${sonarKey}`);
+  }
+
+  const schemaFile = join(METADATA_FOLDER, 'schemas', `${sonarKey}-schema.json`);
+  let schema = '';
+  if (await exists(schemaFile)) {
+    try {
+      schema = `\nimport type { JSONSchema4 } from '@typescript-eslint/utils/json-schema';\nexport const schema = ( ${await readFile(schemaFile, 'utf-8')} ) as const satisfies JSONSchema4;`;
+    } catch {}
+  }
```

wow, impressive one can do an await in `` (backticks) throwing an exception that you swallow..